### PR TITLE
Improve readme to describe semantic rule deduplication and global mecanism

### PR DESCRIPTION
**Pull request by floriangardin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,29 +22,49 @@
...
 and the modelization power of a Random Forest.
 
 
 .. image:: ressources/schema.png
 
 |
 
-One of the original contribution of the package is the use of a semantic deduplication tree.
-It ensures that rules are diversified by design :
+The previous figure explains that skope-rules trains a tree based bagging estimator
```
Comment (ngoix) [inline README.md]: As illustrated in the above figure, the idea behind skope-rules is to fit a tree based ...

File: README.md [hunk 2]
```diff
@@ -22,29 +22,49 @@
...
 
 .. image:: ressources/schema.png
 
 |
 
-One of the original contribution of the package is the use of a semantic deduplication tree.
-It ensures that rules are diversified by design :
+The previous figure explains that skope-rules trains a tree based bagging estimator 
+for a binary classification problem. It then
+extract and filter rules from this estimator based on minimum precision/recall
```
Comment (ngoix) [inline README.md]: A set of rules is then extracted from this tree ensemble, and rules are filtered using thresholds on precision and recall.

File: README.md [hunk 3]
```diff
@@ -22,29 +22,49 @@
...
+
+One of the original contribution of the package is the optional use of another layer of filtering. 
+It uses a *semantic rule deduplication* module which ensures that rules are diversified by design :
 
 |
 
 .. image:: ressources/semantic_deduplication_tree_schema.png
 
-See the `AUTHORS.rst <AUTHORS.rst>`_ file for a list of contributors.
+As described in the previous figure the deduplication algorithm construct a ternary tree
```
Comment (ngoix) [inline README.md]: constructs

File: README.md [hunk 4]
```diff
@@ -22,29 +22,49 @@
...
+One of the original contribution of the package is the optional use of another layer of filtering. 
+It uses a *semantic rule deduplication* module which ensures that rules are diversified by design :
 
 |
 
 .. image:: ressources/semantic_deduplication_tree_schema.png
 
-See the `AUTHORS.rst <AUTHORS.rst>`_ file for a list of contributors.
+As described in the previous figure the deduplication algorithm construct a ternary tree 
+of depth *max_depth_duplication* that classify rules in subgroups that are semantically related.
```
Comment (ngoix) [inline README.md]: classifies

File: README.md [hunk 5]
```diff
@@ -22,29 +22,49 @@
...
+The tree nodes represent the most represented variables in the rule set remaining in the current node.
```
Comment (ngoix) [inline README.md]: unclear, but I don't know how to explain it better

File: README.md [hunk 6]
```diff
@@ -22,29 +22,49 @@
...
+
+- The rule use the variable and the term that contains it looks like : "variable <= value"
+- The rule use the variable and the term that contains it looks like : "variable > value"
+- We don't use the variable in the rule
+ 
+(Nb : a rule can take two different decision path if it uses the same variables 
+twice with a different symbol).
 
+Each leave defines a set of rules. The algorithm
+just take the best rule in each leave in terms of OOB f1-score.
```
Comment (ngoix) [inline README.md]: selects the best rule ...

**Comment by AlJohri** [comment]
bumping @floriangardin. thanks!

**Comment by ngoix** [comment]
travis failure is unrelated, merging

**Pull request merged**